backend-rust/src/indexer.rs

```diff
@@ -4084,23 +4188,38 @@ impl RestakeEarnings {
                             ELSE delegated_stake
                         END
                 WHERE address = $1
-                RETURNING index, delegated_restake_earnings",
+                RETURNING index, delegated_restake_earnings, delegated_target_baker_id",
             self.account_address.as_ref(),
             self.amount
         )
         .fetch_one(tx.as_mut())
         .await?;
-        // When delegated_restake_earnings is None the account is not delegating, so it
-        // might be baking.
-        if account_row.delegated_restake_earnings.is_none() {
+        if let Some(restake) = account_row.delegated_restake_earnings {
+            // Account is delegating.
+            if restake {
```

Comment:
Why is it relevant that we are restaking?

If we are not restaking, wouldnt a change to the state also have to occur?

Reply:
This database operation `RestakeEarnings` only represents the stake changes and not the change in balance.
When restake earnings is enabled, the rewards are put as staked immediately, so we update the pool total stake. If restake earnings is not enabled we don't have to do any changes for this operation.